test/boost/multishard_mutation_query_test.cc

```diff
@@ -186,7 +186,18 @@ read_partitions_with_generic_paged_scan(distributed<replica::database>& db, sche
 
         const auto pkrange_begin_inclusive = res_builder.last_ckey() && res_builder.last_pkey_rows() < slice.partition_row_limit();
 
-        ranges->front() = dht::partition_range(dht::partition_range::bound(res_builder.last_pkey(), pkrange_begin_inclusive), ranges->front().end());
+        {
+            auto& pr = ranges->front();
+            const auto& last_pkey = res_builder.last_pkey();
+            if (!pkrange_begin_inclusive && pr.end() && pr.end()->value().equal(*s, last_pkey)) {
+                ranges->erase(ranges->begin());
+                if (ranges->empty()) {
+                    break;
+                }
+            } else {
+                pr = dht::partition_range(dht::partition_range::bound(last_pkey, pkrange_begin_inclusive), ranges->front().end());
+            }
+        }
```

Comment:
I think it's better to use subtract() or split() or split_after() here so it's clear what's going on. I didn't really look deep since I'm sure there's some primitive that expresses what we want here, rather than playing with the interval components directly.

Reply:
Agree with myself of three weeks ago.

Reply:
I thought I fixed this. Maybe not in the test, I'll check.

Reply:
I did fix it, this is not how the code looks like anymore, I am now using `trim_front()`. Not sure what's up with Github. I rebased and force-pushed again just in case.

Reply:
I did see a trim_front() IIRC.

Reply:
Sorry if I confused you.

Reply:
Ah, so you meant "I agree with myself.. the code looks better this way", not "I agree with myself.. I still think the comment should be fixed".